### PR TITLE
add functionality to calculator

### PR DESCRIPTION
**Pull request by gersonhiggins**
merging "fetaure/events" to "develop"
Add calculare.js 
  add: addition, subtract, divide, multiply and percentage functionality to calculator
checkout linters

### CODE REVIEW

File: src/modules/calculate.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const numberClick = (e) => {
```
Comment (Zuhaib042) [inline src/modules/calculate.js]: - [ ] Kindly consider importing `operate.js` file here. there is no doubt that your app is working fine but we must adhere to the guidelines and requirements mentioned in the project requirements. 
please consider following the guidelines mentioned in the project requirements. you may need to follow the flow as described below:

Project requirements
To help you concentrate on how to work with React events, we have developed the math logic you need to make the Calculator component work. You can use the files [calculate.js](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/calculate.js) and [operate.js](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js).

Create a new directory called logic inside your project
Copy the files calculate.js and operate.js into your logic/ directory
Install [big.js](https://www.npmjs.com/package/big.js) using NPM
Analyze the files calculate.js and operate.js, be sure you understand what they do
Import calculate.js in your Calculator.js component
Implement click event handlers using the math logic from calculate.js and operate.js in your React component
The goal is to make all calculator buttons work according to their functionality e.g. the - button should subtract a given value

File: src/components/calculator.js [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
 const Calculator = () => (
   <div className="calculator">
     <h2>Lets do some math </h2>
     <div className="calculator-box">
       <form>
         <div className="grid">
-          <input type="text" value="0" readOnly />
+          <input className="input" type="text" value="0" readOnly />
+          <p className="saved">0</p>
           <div className="buttons">
```
Comment (Zuhaib042) [inline src/components/calculator.js]: - [ ] As per requirements:
- Note: Practice what you have learned until now: use useState and props.
, kindly consider using `useState` in this Calculator component for representing changes in the output by mathematic operations. you can implement this logic in your component like below:
```
  const [state, setState] = useState({
    total: 0,
    next: null,
    operation: null,
  });

  const handleClick = (event) => {
    const result = calculate(state, event.target.innerText);
    return setState(result);
  };
```

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @gersonhiggins 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✅ Descriptive PR
✅ Professional Readme

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by abdulhamiid — APPROVED** [review]
# :green_circle:  Approved! :partying_face:
Hi @gersonhiggins 👋, 
image
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🌟

- ✔️ Use `useState` and `props`.
- ✔️ Linters checks are passing
- ✔️ Use Git Flow

## Optional suggestions 🤷‍♂️
Your commit messages 👀
image

- [ ] Please let's always use descriptive commit messages - each commit should be small enough to describe it in one sentence. The commit message should be written in the imperative tense and capitalized. [Read more about how to write descriptive commit messages](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8).

Here are some examples of descriptive commits:
image

- [ ] **IMPORTANT: as it is not easy to change commit history, kindly correct this behavior in the future.** 🤗💪

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**